docs/guides/developer/docs/api/workflow-api.md

```diff
@@ -0,0 +1,577 @@
+[TOC]
+# General
+
+The Workflow API is available since API version 1.1.0.
+
+# Workflow instances
+
+### GET /api/workflows
+
+Returns a list of workflow instances.
+
+The following query string parameters are supported to filter, sort and pagingate the returned list:
+
+Query String Parameter | Type                        | Description
+:----------------------|:----------------------------|:-----------
+`filter`               | [`string`](types.md#basic)  | A comma-separated list of filters to limit the results with (see [Filtering](usage.md#filtering)). See the below table for the list of available filters
+`sort`                 | [`string`](types.md#basic)  | A comma-separated list of sort criteria (see [Sorting](usage.md#sorting)).  See the below table for the list of available sort criteria
+`limit`                | [`integer`](types.md#basic) | The maximum number of results to return (see [Pagination](usage.md#pagination))
+`offset`               | [`integer`](types.md#basic) | The index of the first result to return (see [Pagination](usage.md#pagination))
+
+The following filters are available:
+
+Filter Name                      | Description
+:--------------------------------|:-----------
+`state`                          | Workflow instances that are in this state. Can be included multiple times (inclusive or).
+`state_not`                      | Workflow instances that are not in this state. Can be included multiple times (inclusive or).
+`current_operation`              | Workflow instances that are currently executing this operation. Can be included multiple times (inclusive or).
+`current_operation_not`          | Workflow instances that are currently not executing this operation. Can be included multiple times (inclusive or).
+`workflow_definition_identifier` | Workflow instances that use this workflow definition
+`event_identifier`               | Workflow instances where the identifier of the processed event matches
+`event_title`                    | Workflow instances where the title of the processed event matches
```

Comment:
Just wondering... does the workflow need to be that tightly coupled to the event metadata?

Reply:
I have no strong feelings about this. I just used all the available filters and sorting keys of workflow queries. Given that this is implemented in the workflow service, maybe there are use cases where this is necessary? Although, I cannot think of any right now.

Reply:
I think that is a legacy of Matterhorn 1.x: The old admin UI displayed workflow instances (instead of events). Those filters and sort options made it possible to filter/sort workflow instances in an "event-like manner".

If you don't need this for whatever you are working on this PR, I suggest to omit it. I also suggest to bring this up at the technical meeting tomorrow to see what others think...